test/plugin/base_test.rb

```diff
@@ -338,6 +338,81 @@ def test_structured_payload_log
     end
   end
 
+  def test_stackdriver_trace_format_compliant_trace
+    full_stackdriver_trace = \
+      'projects/some-proj/traces/1234567890abcdef1234567890abcdef'
+
+    [
+      {
+        config: APPLICATION_DEFAULT_CONFIG
+      },
+      {
+        config: 'enable_stackdriver_trace_autoformat true'
+      },
+      {
+        config: 'enable_stackdriver_trace_autoformat false'
+      }
+    ].each do |test_params|
+      new_stub_context do
+        setup_gce_metadata_stubs
+        setup_logging_stubs do
+          d = create_driver(test_params[:config])
+          d.emit(DEFAULT_TRACE_KEY => full_stackdriver_trace)
+          d.run
+
+          verify_log_entries(1, COMPUTE_PARAMS, 'jsonPayload') do |entry|
+            assert_equal full_stackdriver_trace, entry['trace']
```

Comment:
`assert_equal` can take a third parameter as the error message when the assertion fails. In order to help with debugging, we could something like:

```
assert_equal full_stackdriver_trace, entry['trace'],
             "test_params #{test_params} failed with entry #{entry}."
```

Same for below.